app/shared/src/commonMain/kotlin/ui/subject/episode/EpisodeViewModel.kt

```diff
@@ -339,6 +345,34 @@ private class EpisodeViewModelImpl(
     override val videoScaffoldConfig: VideoScaffoldConfig by settingsRepository.videoScaffoldConfig
         .flow.produceState(VideoScaffoldConfig.Default)
 
+    fun addDanmakuRegexFilter(filter: DanmakuRegexFilter) {
+        launchInBackground { danmakuRegexFilterRepository.add(filter) }
+    }
+
+    fun editDanmakuRegexFilter(id: String, new: DanmakuRegexFilter) {
+        launchInBackground { danmakuRegexFilterRepository.update(id, new) }
+    }
+
+    fun removeDanmakuRegexFilter(filter: DanmakuRegexFilter) {
+        launchInBackground {
+            danmakuRegexFilterRepository.remove(filter)
+        }
+    }
+
+    fun switchDanmakuRegexFilter(filter: DanmakuRegexFilter) {
+        launchInBackground {
+            danmakuRegexFilterRepository.update(filter.id, filter.copy(enabled = !filter.enabled))
+        }
+    }
+    
+    override val danmakuRegexFilterState = DanmakuRegexFilterState(
+        danmakuRegexFilterList = danmakuRegexFilterRepository.flow.produceState(emptyList()),
+        addDanmakuRegexFilter = ::addDanmakuRegexFilter,
+        editDanmakuRegexFilter = ::editDanmakuRegexFilter,
+        removeDanmakuRegexFilter = ::removeDanmakuRegexFilter,
+        switchDanmakuRegexFilter = ::switchDanmakuRegexFilter,
+    )
```

Comment:
```suggestion
    fun editDanmakuRegexFilter(id: String, new: DanmakuRegexFilter) {
        launchInBackground { danmakuRegexFilterRepository.update(id, new) }
    }

    fun removeDanmakuRegexFilter(filter: DanmakuRegexFilter) {
        launchInBackground {
            danmakuRegexFilterRepository.remove(filter)
        }
    }

    fun switchDanmakuRegexFilter(filter: DanmakuRegexFilter) {
        launchInBackground {
            danmakuRegexFilterRepository.update(filter.id, filter.copy(enabled = !filter.enabled))
        }
    }
    
    override val danmakuRegexFilterState = DanmakuRegexFilterState(
        danmakuRegexFilterList = danmakuRegexFilterRepository.flow.produceState(emptyList()),
        addDanmakuRegexFilter = { launchInBackground { danmakuRegexFilterRepository.add(it) } },
        editDanmakuRegexFilter = ::editDanmakuRegexFilter,
        removeDanmakuRegexFilter = ::removeDanmakuRegexFilter,
        switchDanmakuRegexFilter = ::switchDanmakuRegexFilter,
    )
```

其他几个也改一下. 

这种情况一般就地写 lambda, 更简短, 只有 lambda 的内容很大的时候才会抽一个函数